src/modules/crud.js

```diff
@@ -1,4 +1,151 @@
+  }
+
+  // Refreshes the tasks presented in the UI
+  render() {
+    this.sortTasks();
+    this.todoListBox.innerHTML = '';
+
+    this.todoTask.forEach((task, i) => {
+      const todoTaskElement = document.createElement('li');
+      todoTaskElement.classList.add('todo-task-container');
+
+      const todoTaskContent = document.createElement('div');
+      todoTaskContent.classList.add('task');
+
+      const checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
+
+      todoTaskContent.appendChild(checkbox);
+
+      const description = document.createElement('span');
+      description.classList.add('description');
+      description.contentEditable = 'true';
+      description.textContent = task.description;
+      description.addEventListener('input', () => {
+        this.editTask(i, description.textContent);
+      });
+      todoTaskContent.appendChild(description);
+
+      const deleteIcon = document.createElement('span');
+      deleteIcon.innerHTML = 'delete';
+      deleteIcon.classList.add('material-icons', 'trash-icon');
+      deleteIcon.style.display = 'none';
+      deleteIcon.dataset.task = i;
+
+      const moreVertIcon = document.createElement('span');
+      moreVertIcon.innerHTML = 'more_vert';
+      moreVertIcon.classList.add('material-icons', 'dots-icon');
+
+      todoTaskElement.append(todoTaskContent, deleteIcon, moreVertIcon);
+
+      // Add the task element to the to-do list.
+      this.todoListBox.appendChild(todoTaskElement);
+
+      todoTaskElement.addEventListener(
+        'focus',
+        (event) => {
+          event.currentTarget.querySelector('.dots-icon').style.display = 'none';
+          event.currentTarget.querySelector('.trash-icon').style.display = '';
+          event.currentTarget.style.backgroundColor = 'lightyellow';
+        },
+        true,
+      );
+
+      todoTaskElement.addEventListener(
+        'blur',
+        (event) => {
+          event.currentTarget.querySelector('.dots-icon').style.display = '';
+          event.currentTarget.querySelector('.trash-icon').style.display = 'none';
+          event.currentTarget.style.backgroundColor = '';
+        },
+        true,
+      );
+    });
+
+    // Update local storage.
+    this.updateLocalStorage();
+  }
+
+  // Add new task with the given description, updates local storage and the UI.
+  addTask(description) {
+    if (description.trim() !== '') {
+      const newTask = {
+        description,
+        completed: false,
+        index: this.todoTask.length + 1,
+      };
+      this.todoTask.push(newTask);
+      this.updateLocalStorage();
+      this.render();
+    }
+  }
+
+  // Delete task with matching id, updates local storage and UI.
+  deleteTask(taskId) {
+    this.todoTask.splice(taskId, 1);
+    this.todoTask.forEach((task, index) => task.index === index);
+    this.updateLocalStorage();
+    this.render();
+  }
+
```

Comment:
- [ ] Please make sure that, deleting a task will update all remaining items' indexes, so they represent the current list order and are unique. Currently, as you can see below :point_down: After I deleted a task, your localStorage was not updated to remain items' indexes, so they represent the current list order :blush: 
  
  [Screencast from 07-20-2023 09:05:18 PM.webm](https://github.com/GhausAliKhan/Todo-App/assets/92303891/014d548d-0483-4635-a0c1-4043289049c6)